intro-to-git/intro-to-git-vocab-and-concepts.md

```diff
@@ -0,0 +1,127 @@
+
+When we compare different versions of code to each other, we see how the code shrinks, grows, and changes between them.
+
+In a sense, version control systems let us understand that software is built over time.
+
+In order to use Git to compare versions, sometimes we will need to tell Git to _track_ different files in a project. This is Git's way of saying that it needs to watch for changes in certain files. Of course we will want Git to watch for changes; that's how we get all the benefits!
+
+### Commits are a Snapshot of History
+
+**A commit** is a representation of a version of code at one point of time.
+
+We will build commits as we build software; we will create commits, and over time, these commits will represent different versions of code and build a history.
+
+Commits not only represent a version of code, but also they contain a lot of extra information. Every commit can also contain the following information:
+
+- A summary of lines of code that were added, deleted, or changed, compared to the last commit
+- A way to refer to the last commit, or its "parent" commit
+- A unique ID for the commit, often called the commit hash
+- A commit message, which will be a string that describes the commit's contents
+- The date and time that the commit was made
+
+Large projects will be made of hundreds or thousands of commits.
+
+## Teams Collaborate by Merging Histories
+
+In a work day, we want any number of programmers to be able to write code, contribute commits, and build software.
+
+Git is a version control system that allows for team collaboration. How?
+
+Collaboration in this sense means that everyone can contribute changes to code, and look at those changes of code.
+
+Every codebase that a team will collaborate on will have a **repository,** or a project folder that contains the files, versioned with Git. (This repository will likely be hosted on github.com)
+
+Every developer who wants to contribute code will need access to the code and files, the repo's commit history, and a way to reference the team's repo. In order to get these three things, every developer will **clone that repo** onto their own machine.
+
+When a developer clones the repo onto their machine, it will create a project folder on their computer, with all of the files inside of it. Then, each developer will write code, run tests, build features, **get updates from the team**, and **make commits.**
+
+Then, each dev who has built commits will send the commits to the team's repository.
+
+Recall that each commit includes this information:
+
+- A summary of lines of code that were added, deleted, or changed, compared to the last commit
+- A way to refer to the last commit/its "parent" commit
+
+Using that information, Git (the tool) will go through a series of merging steps to create a unified history of commits.
+
+Now, with a unified history of commits, the repo's history will grow. Software development with Git is a cycle of this process.
```

Comment:
Hmmmm. I also want to get rid of "(the tool)", but I _do_ want to include an alternative. When programmers say "Git does this," "Git does that," a newbie doesn't really get what Git _is_, that it can _do_ things, and end up personifying it as a weird person with its own personality in their head.

I want to get rid of personification/mystification of things like Git, because then people add personality traits to it, "Git is finicky" or "Git doesn't like me." -- I'd like a generation of developers that go "Git is a tool, and if it's not doing what I want it to do, it's because I'm not using the tool correctly"

Maybe to summarize:
Git is a _tool_, and I like emphasizing that often times, tools run whole processes... not because it has a personality, but because that's how the tool is intended to be used.

```suggestion
Using that information, Git must go through a series of merging steps to create a unified history of commits. As each commit is merged in, the repo's history grows. Software development with Git is a cycle of this process.
```

- Changed the verb to "must" to imply a lack of will
- Made it one paragraph -- "Software development with Git is a cycle of this process." on its own line has more of a "mic drop" feel to it than I want lol, and in this case, I'm hoping that "a cycle of this process" is tied to the idea of growing a unified history.

Reply:
sorry for a weirdly poetic response, hopefully it was amusing!

Reply:
Must is a great word choice there!  I confess I don't have the same reaction to the single line "mic drop" 😆, but rolling it into a single paragraphs still reads fine.

I think the goal of avoiding tool personification is going to be a tall order.  I don't know of a single developer who doesn't imbue their tech stack with its own sense of agency.  We know they're all written by other _people_, but by giving them a personality, it gives us room to explain the parts we don't yet understand.  I find git particularly chatty in how it often recommends other commands you might try, and yet very opaque at the same time.  It sometimes requires an act of will to remind myself that the software is just software.  But I think that's something that will need to be reinforced with students repeatedly over time.

In this situation, though, I think the additional changes really well.